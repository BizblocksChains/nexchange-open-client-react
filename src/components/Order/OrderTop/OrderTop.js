import React, { Component } from 'react';
import Bookmark from './Bookmark/Bookmark';
import { I18n } from 'react-i18next';
import styles from './OrderTop.scss';

class OrderTop extends Component {
  constructor(props) {
    super(props);

    this.state = {
      showBookmarkModal: false,
    };
  }

  render() {
    return (
      <I18n ns="translations">
        {t => (
          <div>
            <div className="col-xs-12">
              <h3 className={styles.ref}>
                {t('order.reference')}: <b>{this.props.order.unique_reference}</b>
              </h3>

              <button
                type="button"
                className={`${styles.bookmark} btn btn-default btn-simple`}
                onClick={() => this.setState({ showBookmarkModal: true })}
              >
                {t('bookmark.0')}
              </button>
            </div>

            <Bookmark show={this.state.showBookmarkModal} onClose={() => this.setState({ showBookmarkModal: false })} />
          </div>
        )}
      </I18n>
    );
  }
}

export default OrderTop;
