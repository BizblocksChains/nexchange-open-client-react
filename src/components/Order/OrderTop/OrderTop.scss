.bookmark.bookmark.bookmark.bookmark {
  font-size: 16px;
  padding: 0;
  margin-bottom: 16px;
  color: $green;

  @media (min-width: $screen-ms-min) {
    float: right;
    margin: 32px 0 0 0;
  }

  &:hover,
  &:focus,
  &:active {
    color: $green;
  }
}

.ref.ref {
  margin: 24px 0 0 0;
  color: $black;

  @media (min-width: $screen-ms-min) {
    float: left;
    margin-bottom: 24px;
  }
}
