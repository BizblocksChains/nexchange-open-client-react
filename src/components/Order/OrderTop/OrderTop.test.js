import React from 'react';
import { shallow } from 'enzyme';
import OrderTop from './OrderTop';
import order from 'Mocks/order';

describe('OrderTop', () => {
  it('renders correctly', () => {
    const wrapShallow = shallow(<OrderTop order={order} />).dive();
    expect(wrapShallow).toMatchSnapshot();
  });
});
