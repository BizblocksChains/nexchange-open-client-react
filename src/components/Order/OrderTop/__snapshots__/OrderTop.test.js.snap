// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderTop renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <I18n
    ns="translations"
>
    [Function]
</I18n>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="col-xs-12"
>
          <h3
                    className="ref"
          >
                    order.reference
                    : 
                    <b>
                              OV2O6C
                    </b>
          </h3>
          <button
                    className="bookmark btn btn-default btn-simple"
                    onClick={[Function]}
                    type="button"
          >
                    bookmark.0
          </button>
</div>,
        <Bookmark
          onClose={[Function]}
          show={false}
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3
              className="ref"
>
              order.reference
              : 
              <b>
                            OV2O6C
              </b>
</h3>,
            <button
              className="bookmark btn btn-default btn-simple"
              onClick={[Function]}
              type="button"
>
              bookmark.0
</button>,
          ],
          "className": "col-xs-12",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "order.reference",
                ": ",
                <b>
                  OV2O6C
</b>,
              ],
              "className": "ref",
            },
            "ref": null,
            "rendered": Array [
              "order.reference",
              ": ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "OV2O6C",
                },
                "ref": null,
                "rendered": "OV2O6C",
                "type": "b",
              },
            ],
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "bookmark.0",
              "className": "bookmark btn btn-default btn-simple",
              "onClick": [Function],
              "type": "button",
            },
            "ref": null,
            "rendered": "bookmark.0",
            "type": "button",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "onClose": [Function],
          "show": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="col-xs-12"
>
            <h3
                        className="ref"
            >
                        order.reference
                        : 
                        <b>
                                    OV2O6C
                        </b>
            </h3>
            <button
                        className="bookmark btn btn-default btn-simple"
                        onClick={[Function]}
                        type="button"
            >
                        bookmark.0
            </button>
</div>,
          <Bookmark
            onClose={[Function]}
            show={false}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3
                className="ref"
>
                order.reference
                : 
                <b>
                                OV2O6C
                </b>
</h3>,
              <button
                className="bookmark btn btn-default btn-simple"
                onClick={[Function]}
                type="button"
>
                bookmark.0
</button>,
            ],
            "className": "col-xs-12",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "order.reference",
                  ": ",
                  <b>
                    OV2O6C
</b>,
                ],
                "className": "ref",
              },
              "ref": null,
              "rendered": Array [
                "order.reference",
                ": ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "OV2O6C",
                  },
                  "ref": null,
                  "rendered": "OV2O6C",
                  "type": "b",
                },
              ],
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "bookmark.0",
                "className": "bookmark btn btn-default btn-simple",
                "onClick": [Function],
                "type": "button",
              },
              "ref": null,
              "rendered": "bookmark.0",
              "type": "button",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onClose": [Function],
            "show": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
